test/e2e/tests/test_table.py

```diff
@@ -155,3 +267,391 @@ def test_enable_ttl(self, forum_table):
+        }
+
+        # Patch k8s resource
+        k8s.patch_custom_resource(ref, cr)
+
+        table.wait_until(
+            table_name,
+            table.class_matcher("STANDARD"),
+            timeout_seconds=180,
+            interval_seconds=3,
+        )
+
+    def test_simple_create_gsi(self, office_table):
+        (ref, res) = office_table
+
+        table_name = res["spec"]["tableName"]
+
+        # Check DynamoDB Table exists
+        assert self.table_exists(table_name)
+
+        table.wait_until(
+            table_name,
+            table.gsi_matches([new_gsi_dict("office-per-city", "City", 5)]),
+            timeout_seconds=60*10,
+            interval_seconds=15,
+        )
+
+    def test_create_multi_gsi(self, office_table):
+        (ref, res) = office_table
+
+        table_name = res["spec"]["tableName"]
+
+        # Check DynamoDB Table exists
+        assert self.table_exists(table_name)
+
+        # Get CR latest revision
+        cr = k8s.wait_resource_consumed_by_controller(ref)
+
+        # Creating two more GSIs
+        cr["spec"] = {
+            "attributeDefinitions": [
+                {
+                    "attributeName": "OfficeName",
+                    "attributeType": "S"
+                },
+                {
+                    "attributeName": "Rank",
+                    "attributeType": "S"
+                },
+                {
+                    "attributeName": "City",
+                    "attributeType": "S"
+                },
+                {
+                    "attributeName": "Country",
+                    "attributeType": "S"
+                },
+                {
+                    "attributeName": "State",
+                    "attributeType": "S"
+                },
+            ],
+            "globalSecondaryIndexes": [
+                new_gsi_dict("office-per-city", "City", 5),
+                new_gsi_dict("office-per-country", "Country", 5),
+                new_gsi_dict("office-per-state", "State", 5),
+            ]
+        }
+
+        # Patch k8s resource
+        k8s.patch_custom_resource(ref, cr)
+
+        table.wait_until(
+            table_name,
+            table.gsi_matches(
+                [
+                    new_gsi_dict("office-per-city", "City", 5),
+                    new_gsi_dict("office-per-country", "Country", 5),
+                    new_gsi_dict("office-per-state", "State", 5),
+                ],
+            ),
+            timeout_seconds=60*20,
+            interval_seconds=15,
+        )
+
+    def test_create_delete_gsi(self, office_table):
+        (ref, res) = office_table
+
+        table_name = res["spec"]["tableName"]
+
+        # Check DynamoDB Table exists
+        assert self.table_exists(table_name)
+
+        # Get CR latest revision
+        cr = k8s.wait_resource_consumed_by_controller(ref)
+
+        # Creating two more GSIs
+        # Delete office-per-city and add office-per-country
+        cr["spec"] = {
```

Comment:
Ditto -- are you intending to completely overwrite the spec here?  Why not just update selectively?